api/api.go

```diff
@@ -79,7 +89,8 @@ func (a *api) setParameter(w http.ResponseWriter, r *http.Request) {
 
 	err := a.d.SetParameter(param, value)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
we can introduce a new helper function instead, e.g.
```
func errorHandler(w http.ResponseWriter, err error) {
    w.WriteHeader(http.StatusInternalServerError)
    fmt.Fprintf(w, "Error: %s", err)
}
```

Also, has this been tested, from memory when WriteHeader is called you can't write the response stream anymore, might need to switch it to 

```
fmt.Fprintf(w, "Error: %s", err)
w.WriteHeader(http.StatusInternalServerError)
```

Reply:
I would say that it is exactly the opposite. When `fmt.Fprintf(w, "Error: %s", err)` is before `w.WriteHeader(http.StatusInternalServerError)` it looks like the header is not being changed - it contians default value - `http.StatusOK`

Reply:
If it comes to `errorHandler` that's good idea, I have just added it.

Reply:
thanks @KacperKlys1 , yes my bad, the original code is the way to go.
```
    w.WriteHeader(http.StatusInternalServerError)
    fmt.Fprintf(w, "Error: %s", err)
 ```